engine/include/cubos/engine/ui/canvas/draw_list.hpp

```diff
@@ -0,0 +1,158 @@
+/// @file
+/// @brief Class @ref cubos::engine::UIDrawList.
+/// @ingroup ui-canvas-plugin
+
+#pragma once
+#include <vector>
+
+#include <cubos/core/gl/render_device.hpp>
+
+#include <cubos/engine/api.hpp>
+
+namespace cubos::engine
+{
+    /// @brief Holds a collections of draw commands and their data.
+    /// @ingroup ui-canvas-plugin
+    class CUBOS_ENGINE_API UIDrawList final
+    {
+    public:
+        /// @brief Describes a UI draw call type.
+        /// @ingroup ui-canvas-plugin
+        struct CUBOS_ENGINE_API UIDrawType
```

Comment:
Since these are now internal to the class, you should remove the `UIDraw` prefix.